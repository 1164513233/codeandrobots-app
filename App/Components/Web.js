import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { View, WebView, ActivityIndicator } from 'react-native'

export default class Web extends Component {
  static propTypes = {
    source: PropTypes.string.isRequired
  }

  constructor (props) {
    super(props)
    this.state = { loading: true }
  }

  loadingDone = () => {
    this.setState({ loading: false })
  }

  render () {
    return (
      <View style={{ flex: 1 }}>
        <WebView
          onLoad={this.loadingDone}
          style={{ flex: 1 }}
          source={{uri: this.props.source}} />
        {this.state.loading && (
          <View style={{position: 'absolute', left: 0, right: 0, top: 0, bottom: 0, alignItems: 'center', justifyContent: 'center'}}>
            <ActivityIndicator size='large' />
          </View>
        )}
      </View>
    )
  }
}
