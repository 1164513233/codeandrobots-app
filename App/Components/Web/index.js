import React, { Component } from 'react'
import { View, WebView } from 'react-native'
import PropTypes from 'prop-types'

import { LoadingIndicator } from 'App/Components'

import s from './Styles'

export default class Web extends Component {
  static propTypes = {
    source: PropTypes.string.isRequired
  }

  constructor (props) {
    super(props)
    this.state = { loading: true }
  }

  loadingDone = () => {
    this.setState({ loading: false })
  }

  render () {
    return (
      <View style={s.view}>
        <WebView
          onLoad={this.loadingDone}
          style={s.webView}
          source={{uri: this.props.source}} />
        {this.state.loading && (
          <LoadingIndicator containerStyle={s.loadingView} size='large' />
        )}
      </View>
    )
  }
}
