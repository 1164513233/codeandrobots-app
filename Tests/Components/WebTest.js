import React from 'react'

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Web from 'App/Components/Web'

describe('Screen', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <Web source='http://www.google.com' />)
  })

  it('Component renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
