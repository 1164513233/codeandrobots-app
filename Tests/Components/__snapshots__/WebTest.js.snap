// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen Component renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <WebView
    onLoad={[Function]}
    source={
      Object {
        "uri": "http://www.google.com",
      }
    }
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <ActivityIndicator
      size="large"
    />
  </View>
</View>
`;
